exercises/concept/black-jack/.meta/exemplar.py

```diff
@@ -1,23 +1,66 @@
-def number_of_card(card):
-    if card == 'A':
-        return "ace"
-    elif card == 'J' or card == 'Q' or card == 'K':
-        return 10
+def value_of_card(card):
+    """
+
+    :param card: str - given card.
+    :return: int - value of a given card (face card = 10, pip value otherwise).
+    """
+
+    if card in ['J', 'Q', 'K']:
+        value = 10
     else:
-        return card
+        value = int(card)
+    return value
+
+
+def value_of_ace(hand_value):
+    """
 
+    :param hand_value: int - current hand value.
+    :return: int - value of the upcoming ace card (either 1 or 11).
+    """
 
-def number_of_ace(hand):
-    if hand + 11 <= 21:
-        return 11
+    if hand_value + 11 > 21:
+        value = 1
     else:
-        return 1
+        value = 11
+    return value
+
+
+def is_blackjack(hand):
+    """
+
+    :param hand: list - a pair of cards in hand.
+    :return: bool - if the hand is a blackjack (two cards worth 21).
+    """
 
+    blackjack = False
+    if 'A' in hand:
+        if any(ten_card in hand for ten_card in ['10', 'J', 'Q', 'K']):
+            blackjack = True
+    return blackjack
 
-def blackjack(hand):
-    if 'A' not in hand:
-        return False
-    elif 'J' in hand or 'Q' in hand or 'K' in hand or 10 in hand:
-        return True
+
+def can_split_pairs(hand):
+    """
+
+    :param hand: list - a pair of cards in hand.
+    :return: bool - if the hand can be split into two pairs (i.e. cards are of the same value).
+    """
+
+    card_one, card_two = hand
```

Comment:
Does this depend on a prior `unpacking` concept?

Reply:
Yup.  We can't do multiple assignment or unpacking here, sadly.  In fact, the current prerequisites for this exericse only include `basics`, so `bools` are out too, unless we make them a prerequisite.  

And..the more I look at this, the more issues I see.  We use `lists` here as inputs, but that makes `lists` a prerequisite.  So we have a design doc that's in conflict with where this exercise goes in progression.

Reply:
So.  This was a bad design doc!  😱   At the very least, this exercise will need `numbers`, `bools`, and `conditionals` as prerequisites.  And that's fine.  That's what it should have anyways.
But I am resisting requiring `lists`.  To have a student "wait" until after `lists` to go over `comparisons` is too much.

But if we _don't_ require `lists`, we have a problem with the whole story as it stands.

So I think we need a discussion here on where in the tree it is appropriate to introduce `comparisons` as a concept, and then what is needed to accurately teach it. 

I would contend that this is the appropriate place to teach a first round of `comparisons` -- but that maybe we have a "second round" that goes over `comparisons` in the context of `lists`, `dicts`, `tuples`, etc.

But if we don't require `lists` here, we have to re-work this exercise so that `lists` are not used in the solution.

@pranasziaukas  -- totally understand if this is now not what you signed up for.  But if you **do** want to take on re-thinking this exercise a bit, I am happy to let you do it.


It probably wasn't clear from the issue as logged (or the half-done PR), but all concept exercises can only use the concepts introduced in the exercise itself, or in a prerequisite exercise.  And because this appears early in the "tree", it has very little you can use as a toolset.  That might make this a no-op, or we may have to swap this story for a different story -- like a follow-on to `Currency Exchange`, where we compare different currency valuations.  Or we may need to fold these operators into one of the other concept exercises.

@pranasziaukas  and @IsaacG -- Let me know your thoughts.  Thanks!

Reply:
I guess it's not that big of a deal to avoid `lists` here because the scenario already assumes a pair of cards (instead of an arbitrary number) in relevant tasks.

Reply:
And looks like we can make do with only `numbers`, `bools`, and `conditionals` as prerequisites.